pkg/kube/structured/structured.go

```diff
@@ -235,6 +235,18 @@ func PersistentVolExists(kubeClientset kubernetes.Interface, name, expectedPhase
 	return nil
 }
 
+func PersistentVolClaimExists(kubeClientset kubernetes.Interface, name, expectedPhase string, namespace string) error {
+	vol, err := GetPersistentVolumeClaim(kubeClientset, name, namespace)
+	if err != nil {
+		return err
+	}
+	phase := string(vol.Status.Phase)
+	if phase != expectedPhase {
+		return fmt.Errorf("persistentvolumeclaim had unexpected phase %v, expected phase %v", phase, expectedPhase)
+	}
+	return nil
+}
+
```

Comment:
Is it possible for the claim to exists but have not yet converge to the desire phase? Meaning the claim is now in another phase but will eventually get to the desired phase. If so, retrying here would be worth it. WDYT?

Reply:
yes, will add retry